webapps/ui/cockpit/src/modules/components/TablePaginated/TablePaginated.js

```diff
@@ -0,0 +1,187 @@
+ * @param generateRow is a function which is called for each row, passes the row object and returns a {JSX.Element}
+ * @param fetchPage is a function which returns the page, passes the page number and returns an array of rows (can be a promise); each row must have a unique id
+ * @param fetchRowsCount is a function which returns the total amount of rows wrapped in a promise
+ * @param noRows is {JSX.Element} which is rendered when the rows count is zero
+ * @param onRowSelection is a function which is called when a row is selected and passes the selected rows
+ * @returns the TablePaginated component as {JSX.Element}
+ */
+function TablePaginated({
+  rowsPerPage = 5,
+  header,
+  generateRow,
+  fetchPage,
+  fetchRowsCount = ZERO_NOOP,
+  noRows = null,
+  onRowSelection = NOOP
+}) {
+  const [allSelected, setAllSelected] = useState(false);
+  const [page, setPage] = useState(1);
+  const [rows, setRows] = useState([]);
+  const [rowsCount, setRowsCount] = useState(0);
+  const [rowSelection, setRowSelection] = useState({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (fetchRowsCount !== ZERO_NOOP) {
+      fetchRowsCount().then(rowsCount => setRowsCount(rowsCount.count));
+    } else {
+      setRowsCount(1);
+    }
+  }, [fetchRowsCount]);
+
+  useEffect(() => {
+    setAllSelected(false);
+    setLoading(true);
+
+    Promise.resolve(fetchPage(page)).then(rows => {
+      setLoading(false);
+      setRows(typeof rows === "object" ? Object.values(rows) : rows);
+    });
+  }, [page, fetchPage]);
+
+  useEffect(() => {
+    onRowSelection(rowSelection);
+  }, [onRowSelection, rowSelection]);
+
+  const checkRowSelected = rowId => {
+    return Object.keys(rowSelection).includes(rowId);
+  };
+
+  const select = (e, row) => {
+    if (e.target.checked) {
+      setRowSelection({
+        ...rowSelection,
+        ...{ [row.id]: row }
+      });
+    } else {
+      const { [row.id]: value, ...newRowSelection } = rowSelection;
+      setRowSelection(newRowSelection);
+    }
+  };
+
+  const selectAll = e => {
+    const rowIds = rows.map(row => row.id);
+
+    const checked = e.target.checked;
+    if (checked) {
+      const selectedRows = {};
+      const rowSelectionIds = Object.keys(rowSelection);
+      rows
+        .filter(row => {
+          return !rowSelectionIds.includes(row["id"]);
+        })
+        .forEach(row => {
+          selectedRows[row.id] = row;
+        });
+
+      setRowSelection({
+        ...rowSelection,
+        ...selectedRows
+      });
+    } else {
+      const newRowSelection = {};
+      for (const [rowId, cache] of Object.entries(rowSelection)) {
+        if (!rowIds.includes(rowId)) {
+          newRowSelection[rowId] = cache;
+        }
+      }
+      setRowSelection(newRowSelection);
+    }
+    setAllSelected(checked);
+  };
+
+  if (loading) {
+    return (
+      <div className={classNames("TablePaginated")}>
```

Comment:
`classNames` is not needed here, as we don't add classes based on conditions.

```suggestion
      <div className="TablePaginated">
```